src/doc/rustc/src/jobserver.md

```diff
@@ -0,0 +1,50 @@
+# Jobserver
+
+Internally, `rustc` may take advantage of parallelism. `rustc` will coordinate
+with the build system calling it if a [GNU Make jobserver] is passed in the
+`MAKEFLAGS` environment variable.
```

Comment:
Perhaps it would be good to mention what will happen if a jobserver cannot be found? Maybe something like:

```suggestion
`MAKEFLAGS` environment variable. If a jobserver is not defined, then `rustc` will use a fixed limit on the maximum number of threads.
```

Reply:
Yeah, not sure: I considered doing that, but does `rustc` want to commit to anything here? (i.e. it is always going to be the same?).

In addition, if we mention it, should we say what "fixed limit" means? Should we link to `-Zthreads`, `-Ccodegen-units` etc.?

Reply:
I don't think we want to make any commitments, and I don't think mentioning that `rustc` will "do it's own thing" in the absence of a jobserver is really making a commitment. 

It could be more loosely worded to something like "If a jobserver is not defined, then `rustc` will choose the number of threads to use."  Or whatever wording seems sufficiently vague.

I don't think I would bother mentioning `-Zthreads` for now, since it is highly experimental, and there isn't really any documentation on it (that I am aware of). I also probably wouldn't bother with `codegen-units` since I don't think we want to get too explicit about how `rustc` creates threads.

Just a suggestion, though. I'm fine not mentioning it.

Reply:
Makes sense. The "sufficiently vague" option sounded like a good compromise to me, so I added that one, please take a look ("threads" -> "jobs", "defined" -> "passed").